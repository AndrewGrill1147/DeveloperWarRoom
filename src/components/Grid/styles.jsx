
const styles = {
    fixedToBottom: {
      position: 'fixed',
      bottom: '0',
      right: '0',
      margin: '10px',
    },
    removeStyle: {
      position: 'absolute',
      right: '2px',
      top: 0,
      cursor: 'pointer',
    },
    horizontalHeaderBarStyle: {
      display: 'inline',
      whiteSpace: 'nowrap',
    },
    iconAlignment: {
      position: 'fixed',
      top: '0',
      right: '0',
      marginTop: '8px',
    },
    gridItemPaperStyle: {
      height: '100%',
      width: '100%',
      margin: '0px',
      textAlign: 'left',
      display: 'inline-block',
    },
    menuBarStyle: {
      backgroundColor: 'rgb(0, 188, 212)',
    },
    gridItem: {
      // TODO?
      // overflow: 'auto',
    },
  };

  export default styles;