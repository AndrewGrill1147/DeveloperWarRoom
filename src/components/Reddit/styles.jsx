const styles = {
  primaryText: {
    display: 'block',
    marginLeft: '50px',
    color: '#00bcd4',
  },
  loginButton: {
    marginTop: '5px',
  },
  headerBar: {
    height: '40px',
  },
  refreshIconButton: {
    position: 'absolute',
    top: '0px',
    right: '25px',
    color: 'gray',
  },
  refreshIcon: {
    color: 'gray',
    height: '30px',
    width: '30px',
  },
  secondaryText: {
    display: 'block',
    marginLeft: '50px',
  },
  img: {
    maxWidth: '100%',
  },
  leftIcon: {
    height: 'auto',
    maxHeight: '80px',
    maxWidth: '100px',
    width: 'auto',
    margin: 'auto',
    padding: 'auto',
  },
  tabHeadline: {
    fontSize: 12,
    paddingTop: 2,
    marginBottom: 2,
    fontWeight: 200,
  },
  listStyle: {
    listStyleType: 'none',
    margin: 0,
    padding: 0,
  },
  divStyle: {
    display: 'flex',
    flexDirection: 'column',
    background: '#75a3a3',
    width: '100%',
    height: '100%',
  },
  expand: {
    width: '100%',
    height: 'calc(100% - 40px)',
    maxHeight: 'calc(100% - 40px)',
    overflow: 'auto',
  },
  expand2: {
    width: '100%',
    height: '100%',
    maxHeight: '100%',
  },
  listItemStyle: {
    minHeight: '100px',
    maxHeight: 'auto',
  },
  selectField: {
    width: '150px',
    position: 'absolute',
    top: '0px',
    right: '75px',
  },
};

export default styles;
