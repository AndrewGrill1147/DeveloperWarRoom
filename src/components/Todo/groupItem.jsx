import React, { Component } from 'react';
import { ListItem, TextField, IconButton, IconMenu, MenuItem } from 'material-ui';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';

const ENTER_KEY = 13;
const ESC_KEY = 27;
const iconButtonElement = (
  <IconButton
    touch
  >
    <MoreVertIcon color="gray" />
  </IconButton>
);

class GroupItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      editText: this.props.group.name,
    };
    // edit functionality copied from todoItem
    this.startEdit = this.startEdit.bind(this);
    this.handleInput = this.handleInput.bind(this);
    this.handleEditGroupKeyDown = this.handleEditGroupKeyDown.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    const val = event.target.value.trim();
    if (val) {
      this.props.onSave({ ...this.props.group, ...{ name: val } });
    } else {
      this.props.onDelete();
    }
  }

  startEdit() {
    this.props.onEdit();
    this.setState({ editText: this.props.group.name });
  }

  handleInput(event) {
    this.setState({ editText: event.target.value });
  }

  handleEditGroupKeyDown(event) {
    if (event.which === ESC_KEY) {
      this.setState({ editText: this.props.group.name });
      this.props.onCancel();
    } else if (event.which === ENTER_KEY) {
      this.handleSubmit(event);
    }
  }

  render() {
    const rightIconButtons = (
      <IconMenu
        iconButtonElement={iconButtonElement}
      >
        <MenuItem onClick={this.startEdit}>Edit</MenuItem>
        <MenuItem onClick={this.props.onDelete}>Delete</MenuItem>
      </IconMenu>
    );
    if (!this.props.editing) {
      return (
        <ListItem
          primaryText={this.props.group.name}
          rightIconButton={this.props.group.id !== 0 ? rightIconButtons : null}
        />
      );
    }
    return (
      <ListItem>
        <TextField
          id={this.props.group.id.toString()}
          value={this.state.editText}
          underlineShow
          onInput={this.handleInput}
          onKeyDown={this.handleEditGroupKeyDown}
          autoFocus
          fullWidth
        />
      </ListItem>
    );
  }
}

export default GroupItem;
