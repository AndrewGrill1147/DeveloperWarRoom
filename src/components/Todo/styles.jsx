const styles = {
  tabHeadline: {
    fontSize: 12,
    paddingTop: 2,
    marginBottom: 2,
    fontWeight: 200,
  },
  listStyle: {
    listStyleType: 'none',
    margin: 0,
    padding: 0,
  },
  divStyle: {
    display: 'flex',
    flexDirection: 'column',
    background: '#75a3a3',
    width: '100%',
    height: '100%',
  },
  expand: {
    width: '100%',
    height: '100%',
    maxHeight: '100%',
    overflow: 'auto',
  },
  groupTextField: {
    width: '100%',
    textAlign: 'center',
  },
  newTodoTextField: {
    top: '0px',
    display: 'inline-block',
    verticalAlign: 'middle',
    width: '60%',
  },
  selectGroup: {
    display: 'inline-block',
    verticalAlign: 'middle',
    width: '40%',
  },
  newTodoHint: {
    width: '98%',
    whiteSpace: 'nowrap',
    textOverflow: 'ellipsis',
    overflow: 'hidden',
  },
};

export default styles;
