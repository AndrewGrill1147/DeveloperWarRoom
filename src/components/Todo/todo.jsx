/* eslint-env browser */
/* eslint react/jsx-no-bind: 0 */
import React, { Component } from 'react';
import { List, Subheader, Tabs, Tab, Paper, TextField, Divider } from 'material-ui';
import TodoItem from './todoItem';

const styles = {
  tabHeadline: {
    fontSize: 12,
    paddingTop: 2,
    marginBottom: 2,
    fontWeight: 200,
  },
  listStyle: {
    listStyleType: 'none',
    margin: 0,
    padding: 0,
  },
  divStyle: {
    display: 'flex',
    flexDirection: 'column',
    background: '#75a3a3',
    width: '100%',
    height: '100%',
  },
  expand: {
    width: '100%',
    height: '100%',
    maxHeight: '100%',
    overflow: 'auto',
  },
};

const status = {
  ALL: 'all',
  ACTIVE: 'active',
  COMPLETED: 'completed',
};
const ENTER_KEY = 13;

class Todo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      editing: null,
      newTodo: '',
      todos: [],
      storageKey: this.constructor.name,
    };

    // .bind(this) can be placed here
    this.componentWillUnmount = this.componentWillUnmount.bind(this);
  }

  // Life cycle methods
  componentDidMount() {
    const savedState = localStorage.getItem(this.state.storageKey);
    if (savedState !== null) {
      this.setState(JSON.parse(savedState));
    }
    window.addEventListener('beforeunload', this.componentWillUnmount);
  }

  componentWillUnmount() {
    localStorage.setItem(this.state.storageKey, JSON.stringify(this.state));
    window.removeEventListener('beforeunload', this.componentDidUpdate);
  }

  onDelete(deletedTodo) {
    const updatedTodos = this.state.todos.filter(todo => todo !== deletedTodo);
    this.setState({ todos: updatedTodos });
  }

  onCancel() {
    this.setState({ editing: null });
  }

  onEdit(editTodo) {
    this.setState({ editing: editTodo.id });
  }

  onSave(editedTodo) {
    // update the todos
    const updatedTodos = [...this.state.todos].map((todo) => {
      if (todo.id === editedTodo.id) {
        return editedTodo;
      }
      return todo;
    });

    this.setState({ todos: updatedTodos });
    this.onCancel();
  }

  /* toggledTodo comes from binding, not call back signature */
  onToggle(toggledTodo) {
    const updatedTodos = this.state.todos.map(todo => (todo !== toggledTodo ?
      todo :
      { ...todo, ...{ completed: !todo.completed } }));
    this.setState({ todos: updatedTodos });
  }

  handleNewTodoKeyDown(event) {
    if (event.keyCode !== ENTER_KEY) {
      return;
    }
    event.preventDefault();
    const val = this.state.newTodo.trim();

    if (val) {
      this.addTodo(val);
      this.setState({ newTodo: '' });
    }
  }

  handleInput(event) {
    this.setState({ newTodo: event.target.value });
  }

  addTodo(value) {
    const todo = {
      id: Date.now(),
      title: value,
      completed: false,
    };
    this.setState({ todos: [...this.state.todos, todo] });
  }

  loadState() {
    localStorage.getItem();
    this.alert('loading state');
  }

  saveState() {
    this.alert('Saving state!');
  }

  render() {
    const todosByStatus = {};
    Object.values(status).forEach((s) => {
      todosByStatus[s] = [];
    });

    [...this.state.todos].forEach((todo) => {
      const todoComponent = (
        <TodoItem
          key={todo.id}
          todo={todo}
          editing={this.state.editing === todo.id}
          onEdit={this.onEdit.bind(this, todo)}
          onCancel={this.onCancel.bind(this)}
          onToggle={this.onToggle.bind(this, todo)}
          onDelete={this.onDelete.bind(this, todo)}
          onSave={this.onSave.bind(this)}
        />
      );
      todosByStatus[status.ALL].push(todoComponent);
      todosByStatus[todo.completed ? status.COMPLETED : status.ACTIVE].push(todoComponent);
    }, this);

    // this was going to be a toggle button
    return (
      <div style={styles.divStyle}>
        <Paper zDepth={0} rounded={false} style={styles.expand}>
          <TextField
            id="newTodo"
            value={this.state.newTodo}
            hintText="Todo List - What needs to be done?"
            underlineShow
            onInput={this.handleInput.bind(this)}
            onKeyDown={this.handleNewTodoKeyDown.bind(this)}
            fullWidth
            multiLine
          />

          <Divider />
          <Tabs>
            <Tab label="active">
              <div>
                <List>
                  {todosByStatus[status.ACTIVE]}
                  <Subheader inset={false}>
                    {todosByStatus[status.ACTIVE].length} items
                  </Subheader>
                </List>
              </div>
            </Tab>
            <Tab label="completed">
              <div>
                <List styles={styles.expand}>
                  {todosByStatus[status.COMPLETED]}
                  <Subheader inset={false}>
                    {todosByStatus[status.COMPLETED].length} items
                  </Subheader>
                </List>
              </div>
            </Tab>
            <Tab label="all">
              <div>
                {todosByStatus[status.ALL]}
                <Subheader inset={false}>
                  {todosByStatus[status.ALL].length} items
                </Subheader>
              </div>
            </Tab>
          </Tabs>
        </Paper>
      </div>
    );
  }
}

export default Todo;
