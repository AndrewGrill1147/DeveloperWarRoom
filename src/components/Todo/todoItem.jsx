import React, { Component } from 'react';
import { ListItem, Checkbox, TextField, IconButton, IconMenu, MenuItem, DropDownMenu } from 'material-ui';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';

const iconButtonElement = (
  <IconButton
    touch
    tooltip="more"
    tooltipPosition="bottom-left"
  >
    <MoreVertIcon color="gray" />
  </IconButton>
);

const ENTER_KEY = 13;
const ESC_KEY = 27;

class TodoItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      editText: this.props.todo.title,
    };

    this.startEdit = this.startEdit.bind(this);
    this.handleInput = this.handleInput.bind(this);
    this.handleEditTodoKeyDown = this.handleEditTodoKeyDown.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    const val = event.target.value.trim();
    if (val) {
      this.props.onSave({ ...this.props.todo, ...{ title: val } });
    } else {
      this.props.onDelete();
    }
  }

  startEdit() {
    this.props.onEdit();
    this.setState({ editText: this.props.todo.title });
  }

  handleInput(event) {
    this.setState({ editText: event.target.value });
  }

  handleEditTodoKeyDown(event) {
    if (event.which === ESC_KEY) {
      this.setState({ editText: this.props.title });
      this.props.onCancel();
    } else if (event.which === ENTER_KEY) {
      this.handleSubmit(event);
    }
  }

  render() {
    const rightIconButtons = (
      <IconMenu iconButtonElement={iconButtonElement}>
        <MenuItem onClick={this.startEdit}>Edit</MenuItem>
        <MenuItem onClick={this.props.onDelete}>Delete</MenuItem>
      </IconMenu>
    );

    if (!this.props.editing) {
      return (
        <ListItem
          primaryText={this.props.todo.title}
          leftCheckbox={
            <Checkbox
              onCheck={this.props.onToggle}
              checked={this.props.todo.completed}
            />
          }
          rightIconButton={rightIconButtons}
        />
      );
    }

    // we are editing this element
    return (
      <ListItem>
        <TextField
          id={this.props.todo.id.toString()}
          value={this.state.editText}
          underlineShow
          onInput={this.handleInput}
          onKeyDown={this.handleEditTodoKeyDown}
          autoFocus
          fullWidth
        />
      </ListItem>
    );
  }
}

export default TodoItem;
