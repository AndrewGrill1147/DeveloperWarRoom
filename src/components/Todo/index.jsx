import React, { Component } from 'react';
import { List, Subheader, Tabs, Tab, Paper, TextField, Divider, MenuItem, SelectField } from 'material-ui';
import TodoItem from './todoItem';
import GroupItem from './groupItem';
import LocalStorageAPI from './../../helpers/localstorageAPI';
import styles from './styles';

const status = {
  ALL: 'all',
  ACTIVE: 'active',
  COMPLETED: 'completed',
};
const ENTER_KEY = 13;

class Todo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      editing: null,
      editingGroup: null,
      newTodo: '',
      newGroup: '',
      todos: [],
      currentGroup: 0,
      groupList: [{ id: 0, name: 'default' }],
      storageKey: this.constructor.name,
      currentTab: status.ACTIVE,
    };

    // .bind(this) can be placed here
    this.componentDidUpdate = this.componentDidUpdate.bind(this);
    this.handleGroupChange = this.handleGroupChange.bind(this);

    const savedState = LocalStorageAPI.get(this.state.storageKey);
    this.state.currentGroup = 0;
    if (savedState !== null) {
      this.state = savedState;
    }
  }
  componentDidUpdate() {
    LocalStorageAPI.put(this.state.storageKey, this.state);
  }

  onDelete(deletedTodo) {
    const updatedTodos = this.state.todos.filter(todo => todo !== deletedTodo);
    this.setState({ todos: updatedTodos });
  }

  onCancel() {
    this.setState({ editing: null });
  }

  onEdit(editTodo) {
    this.setState({ editing: editTodo.id });
  }

  onSave(editedTodo) {
    // update the todos
    const updatedTodos = [...this.state.todos].map((todo) => {
      if (todo.id === editedTodo.id) {
        return editedTodo;
      }
      return todo;
    });

    this.setState({ todos: updatedTodos });
    this.onCancel();
  }

  onCancelGroup() {
    this.setState({ editingGroup: null });
  }

  onEditGroup(editGroup) {
    this.setState({ editingGroup: editGroup.id });
  }

  onSaveGroup(editedGroup) {
    const updatedGroups = [...this.state.groupList].map((group) => {
      if (group.id === editedGroup.id) {
        return editedGroup;
      }
      return group;
    });

    this.setState({ groupList: updatedGroups });
    this.onCancelGroup();
  }

  /* toggledTodo comes from binding, not call back signature */
  onToggle(toggledTodo) {
    const updatedTodos = this.state.todos.map(todo => (todo !== toggledTodo ?
      todo :
      { ...todo, ...{ completed: !todo.completed } }));
    this.setState({ todos: updatedTodos });
  }

  addGroup(value) {
    const newGroup = {
      id: Date.now(),
      name: value,
    };
    this.setState({ groupList: [...this.state.groupList, newGroup] });
  }

  deleteGroup(groupId) {
    const updatedGroups = this.state.groupList.filter(group => group.id !== groupId);
    const updatedTodos = this.state.todos.filter(todo => todo.group !== groupId);
    this.setState({ groupList: updatedGroups, todos: updatedTodos });
    if (groupId === this.state.currentGroup) { this.setState({ currentGroup: 0 }); }
  }

  handleNewTodoKeyDown(event) {
    if (event.keyCode !== ENTER_KEY) {
      return;
    }
    event.preventDefault();
    const val = this.state.newTodo.trim();

    if (val) {
      this.addTodo(val);
      this.setState({ newTodo: '' });
    }
  }

  handleNewGroupKeyDown(event) {
    if (event.keyCode !== ENTER_KEY) {
      return;
    }
    event.preventDefault();
    const val = this.state.newGroup.trim();

    if (val) {
      this.addGroup(val);
      this.setState({ newGroup: '' });
    }
  }

  handleInput(event) {
    if (event.target.id === 'newTodo') {
      this.setState({ newTodo: event.target.value });
    } else if (event.target.id === 'newGroup') {
      this.setState({ newGroup: event.target.value });
    }
  }

  addTodo(value) {
    const todo = {
      id: Date.now(),
      title: value,
      completed: false,
      group: this.state.currentGroup,
    };
    this.setState({ todos: [...this.state.todos, todo] });
  }

  handleGroupChange(event, index, value) {
    this.setState({ currentGroup: value });
  }

  render() {
    const todosByStatus = {};
    todosByStatus[status.ACTIVE] = [];
    todosByStatus[status.COMPLETED] = [];
    todosByStatus[status.ALL] = [];

    [...this.state.todos].filter(todo => todo.group === this.state.currentGroup).forEach((todo) => {
      const todoComponent = (
        <TodoItem
          key={todo.id}
          todo={todo}
          editing={this.state.editing === todo.id}
          onEdit={this.onEdit.bind(this, todo)}
          onCancel={this.onCancel.bind(this)}
          onToggle={this.onToggle.bind(this, todo)}
          onDelete={this.onDelete.bind(this, todo)}
          onSave={this.onSave.bind(this)}
        />
      );
      todosByStatus[status.ALL].push(todoComponent);
      todosByStatus[todo.completed ? status.COMPLETED : status.ACTIVE].push(todoComponent);
    }, this);
    // this was going to be a toggle button
    return (
      <div style={styles.divStyle}>
        <Paper zDepth={0} rounded={false} style={styles.expand}>
          <div>
            <TextField
              id="newTodo"
              style={styles.newTodoTextField}
              value={this.state.newTodo}
              hintText="Todo List - What needs to be done?"
              hintStyle={styles.newTodoHint}
              underlineShow
              onInput={this.handleInput.bind(this)}
              onKeyDown={this.handleNewTodoKeyDown.bind(this)}
            />
            <SelectField
              value={this.state.currentGroup}
              onChange={this.handleGroupChange}
              style={styles.selectGroup}
            >
              {this.state.groupList.map(listItem =>
                <MenuItem key={listItem.id} value={listItem.id} primaryText={listItem.name} />)}
            </SelectField>
          </div>

          <Divider />
          <Tabs
            value={this.state.currentTab}
            onChange={value => this.setState({ currentTab: value })}
          >
            <Tab
              value={status.ACTIVE}
              label={status.ACTIVE}
            >
              <div>
                <List>
                  {todosByStatus[status.ACTIVE]}
                  <Subheader inset={false}>
                    {todosByStatus[status.ACTIVE].length} items
                  </Subheader>
                </List>
              </div>
            </Tab>
            <Tab
              value={status.COMPLETED}
              label={status.COMPLETED}
            >
              <div>
                <List>
                  {todosByStatus[status.COMPLETED]}
                  <Subheader inset={false}>
                    {todosByStatus[status.COMPLETED].length} items
                  </Subheader>
                </List>
              </div>
            </Tab>
            <Tab
              value={status.ALL}
              label={status.ALL}
            >
              <div>
                <List>
                  {todosByStatus[status.ALL]}
                  <Subheader inset={false}>
                    {todosByStatus[status.ALL].length} items
                  </Subheader>
                </List>
              </div>
            </Tab>
            <Tab
              value="groups"
              label="groups"
            >
              <List>
                {
                  this.state.groupList.map(group =>
                    (<GroupItem
                      key={group.id}
                      group={group}
                      onDelete={this.deleteGroup.bind(this, group.id)}
                      editing={this.state.editingGroup === group.id}
                      onCancel={this.onCancelGroup.bind(this)}
                      onEdit={this.onEditGroup.bind(this, group)}
                      onSave={this.onSaveGroup.bind(this)}
                      on
                    />))
                }
              </List>
              <TextField
                id="newGroup"
                fullWidth
                hintText={<b>add group</b>}
                hintStyle={styles.groupTextField}
                inputStyle={styles.groupTextField}
                underlineShow
                value={this.state.newGroup}
                onInput={this.handleInput.bind(this)}
                onKeyDown={this.handleNewGroupKeyDown.bind(this)}
              />
            </Tab>
          </Tabs>
        </Paper>
      </div>
    );
  }
}

export default Todo;
