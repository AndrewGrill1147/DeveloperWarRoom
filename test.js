import React from 'react';
import { shallow } from 'enzyme';
import Todo from './src/components/Todo/todo';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MuiThemeProvider } from 'material-ui/styles';


describe('Todo component renders the todo correctly', () => {
    it('renders correctly', () => {
      const rendered = renderer.create(
          <MuiThemeProvider>
        <Todo/>
        </MuiThemeProvider>
      );
      console.log(rendered);
      expect(rendered.toJSON()).toMatchSnapshot();
    });
  });

describe('Math example', () => {

    it('addition', () => {
        expect(2 + 2).toEqual(4);
    })

});