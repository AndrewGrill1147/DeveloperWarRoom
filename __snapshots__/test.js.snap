// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders the todo correctly renders correctly 1`] = `
<div
  style={
    Object {
      "background": "#75a3a3",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": "0px",
        "boxShadow": undefined,
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "Roboto, sans-serif",
        "height": "100%",
        "maxHeight": "100%",
        "muiPrepared": true,
        "overflow": "auto",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": "100%",
      }
    }
  >
    <hr
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      }
    />
    <div
      style={
        Object {
          "muiPrepared": true,
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "#00bcd4",
            "display": "flex",
            "muiPrepared": true,
            "whiteSpace": "nowrap",
            "width": "100%",
          }
        }
      >
        <button
          disabled={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "background": "none",
              "border": 10,
              "boxSizing": "border-box",
              "color": "#ffffff",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 14,
              "fontWeight": 500,
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textDecoration": "none",
              "textTransform": "uppercase",
              "verticalAlign": null,
              "width": "33.333333333333336%",
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "column",
                  "height": 48,
                  "justifyContent": "center",
                }
              }
            >
              active
            </div>
          </div>
        </button>
        <button
          disabled={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "background": "none",
              "border": 10,
              "boxSizing": "border-box",
              "color": "rgba(255, 255, 255, 0.7)",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 14,
              "fontWeight": 500,
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textDecoration": "none",
              "textTransform": "uppercase",
              "verticalAlign": null,
              "width": "33.333333333333336%",
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "column",
                  "height": 48,
                  "justifyContent": "center",
                }
              }
            >
              completed
            </div>
          </div>
        </button>
        <button
          disabled={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "background": "none",
              "border": 10,
              "boxSizing": "border-box",
              "color": "rgba(255, 255, 255, 0.7)",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": 14,
              "fontWeight": 500,
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textDecoration": "none",
              "textTransform": "uppercase",
              "verticalAlign": null,
              "width": "33.333333333333336%",
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "column",
                  "height": 48,
                  "justifyContent": "center",
                }
              }
            >
              all
            </div>
          </div>
        </button>
      </div>
      <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "#ff4081",
              "bottom": 0,
              "display": "block",
              "height": 2,
              "left": "0%",
              "marginTop": -2,
              "muiPrepared": true,
              "position": "relative",
              "transition": "left 1s cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "33.333333333333336%",
            }
          }
        />
      </div>
      <div
        className={undefined}
        style={
          Object {
            "muiPrepared": true,
          }
        }
      >
        <div
          style={
            Object {
              "position": "relative",
              "textAlign": "initial",
              "width": "100%",
            }
          }
        >
          <div>
            <div
              style={
                Object {
                  "muiPrepared": true,
                  "padding": "0px 0px 8px 0px",
                }
              }
            >
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontSize": 14,
                    "fontWeight": 500,
                    "lineHeight": "48px",
                    "muiPrepared": true,
                    "paddingLeft": 16,
                    "width": "100%",
                  }
                }
              >
                0
                 items
              </div>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "height": 0,
              "overflow": "hidden",
              "position": "relative",
              "textAlign": "initial",
              "width": "100%",
            }
          }
        >
          <div>
            <div
              style={
                Object {
                  "muiPrepared": true,
                  "padding": "0px 0px 8px 0px",
                }
              }
              styles={
                Object {
                  "height": "100%",
                  "maxHeight": "100%",
                  "overflow": "auto",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontSize": 14,
                    "fontWeight": 500,
                    "lineHeight": "48px",
                    "muiPrepared": true,
                    "paddingLeft": 16,
                    "width": "100%",
                  }
                }
              >
                0
                 items
              </div>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "height": 0,
              "overflow": "hidden",
              "position": "relative",
              "textAlign": "initial",
              "width": "100%",
            }
          }
        >
          <div>
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontSize": 14,
                  "fontWeight": 500,
                  "lineHeight": "48px",
                  "muiPrepared": true,
                  "paddingLeft": 16,
                  "width": "100%",
                }
              }
            >
              0
               items
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
